docs/architecture/adr-004-genesis-modes.md

```diff
@@ -0,0 +1,253 @@
+# ADR 4: Multiple Genesis modes
+
+## Changelog
+
+- Sept 3, 2019: Initial draft
+
+## Context
+
+In the current SDK, when the `GenesisState` is initialized through a `InitGenesis` call, some of the `GenesisState` fields are loaded from a `genesis.json` file, whereas in other cases they are initialized (_i.e_ populated) if the corresponding object value from JSON is empty, undefined or its value is zero.
+
+This may lead to confusion because as is currently programmed, `InitGenesis`
+initializes _and_ loads the values at the same time. When in reallity they should be
+two separated disctinct processes.
+
+A recurring case of this problem is to check if a balance is empty and initialize it
```

Comment:
This is specific to supply and supply only, yet there is no mention of supply in the context before this. This ADR seems to be directly aimed at irregularities within `x/supply` _only_.

Reply:
I think that this is relevant to other modules as well. Even just using empty or default values for what is held in the `Minter` for instance